amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -446,41 +455,128 @@ void runProcessLoop() {
+            final Set<StreamIdentifier> staleStreamIdsToBeDeleted = staleStreamIdDeletionDecisionMap.get(false).stream()
+                    .filter(streamIdentifier ->
+                            Duration.between(staleStreamDeletionMap.get(streamIdentifier), Instant.now()).toMillis()
+                                    >= getOldStreamDeferredDeletionPeriodMillis()).collect(Collectors.toSet());
+            streamsSynced.addAll(deleteMultiStreamLeases(staleStreamIdsToBeDeleted));
+
+            // Purge the active streams from stale streams list.
+            final Set<StreamIdentifier> staleStreamIdsToBeRevived = staleStreamIdDeletionDecisionMap.get(true);
+            removeActiveStreamsFromStaleStreamsList(staleStreamIdsToBeRevived);
+
             streamSyncWatch.reset().start();
         }
         return streamsSynced;
     }
 
+    @VisibleForTesting
+    long getOldStreamDeferredDeletionPeriodMillis() {
+        return OLD_STREAM_DEFERRED_DELETION_PERIOD_MILLIS;
+    }
+
     @VisibleForTesting
     boolean shouldSyncStreamsNow() {
         return isMultiStreamMode && (streamSyncWatch.elapsed(TimeUnit.MILLISECONDS) > NEW_STREAM_CHECK_INTERVAL_MILLIS);
     }
 
-    private void syncStreamsFromLeaseTableOnAppInit()
+    private void syncStreamsFromLeaseTableOnAppInit(List<MultiStreamLease> leases) {
+        final Set<StreamIdentifier> streamIdentifiers = leases.stream()
+                .map(lease -> StreamIdentifier.multiStreamInstance(lease.streamIdentifier()))
+                .collect(Collectors.toSet());
+        for (StreamIdentifier streamIdentifier : streamIdentifiers) {
+            if (!currentStreamConfigMap.containsKey(streamIdentifier)) {
+                currentStreamConfigMap.put(streamIdentifier, getDefaultStreamConfig(streamIdentifier));
+            }
+        }
+    }
+
+    private List<MultiStreamLease> fetchMultiStreamLeases()
             throws DependencyException, ProvisionedThroughputException, InvalidStateException {
-        if (!leasesSyncedOnAppInit && isMultiStreamMode) {
-            final Set<StreamIdentifier> streamIdentifiers = leaseCoordinator.leaseRefresher().listLeases().stream()
-                    .map(lease -> StreamIdentifier.multiStreamInstance(((MultiStreamLease) lease).streamIdentifier()))
-                    .collect(Collectors.toSet());
-            for (StreamIdentifier streamIdentifier : streamIdentifiers) {
-                if (!currentStreamConfigMap.containsKey(streamIdentifier)) {
-                    currentStreamConfigMap.put(streamIdentifier, getDefaultStreamConfig(streamIdentifier));
+        return (List<MultiStreamLease>) ((List) leaseCoordinator.leaseRefresher().listLeases());
+    }
+
+    private void removeActiveStreamsFromStaleStreamsList(Set<StreamIdentifier> streamIdentifiers) {
+        for(StreamIdentifier streamIdentifier : streamIdentifiers) {
+            staleStreamDeletionMap.remove(streamIdentifier);
+        }
```

Comment:
This function name doesn't align with what it is doing. It is removing from the `staleStreamDeletionMap`, and there is no check within the function that the streams removed are active. Instead, let's change the signature to `removeFromStaleStreamsList(Set<StreamIdentifier> activeStreams)` so it is clear that the expectation is that any stream passed in is removed, and the passed in streams are expected to be active.

_For example, given the name here i would expect the function to find the active streams, and remove them._

Reply:
Sounds good. Renaming the method.